server/http.go

```diff
@@ -192,30 +212,61 @@ func respondJSON(w http.ResponseWriter, obj interface{}) (int, error) {
 func (p *Plugin) respondTemplate(w http.ResponseWriter, r *http.Request, contentType string, values interface{}) (int, error) {
 	_, path := splitInstancePath(r.URL.Path)
 	w.Header().Set("Content-Type", contentType)
-	t := p.templates[path]
+	if contentType == ContentTypeHTML {
+		t := p.htmlTemplates[path]
+		if t == nil {
+			return respondErr(w, http.StatusInternalServerError,
+				errors.New("no template found for "+path))
+		}
+
+		if err := t.Execute(w, values); err != nil {
+			return http.StatusInternalServerError, errors.WithMessage(err, "failed to write response")
+		}
+
+		return http.StatusOK, nil
+	}
+
+	t := p.textTemplates[path]
 	if t == nil {
 		return respondErr(w, http.StatusInternalServerError,
 			errors.New("no template found for "+path))
 	}
-	err := t.Execute(w, values)
-	if err != nil {
+
+	if err := t.Execute(w, values); err != nil {
 		return http.StatusInternalServerError, errors.WithMessage(err, "failed to write response")
 	}
+
 	return http.StatusOK, nil
 }
 
 func (p *Plugin) respondSpecialTemplate(w http.ResponseWriter, key string, status int, contentType string, values interface{}) (int, error) {
 	w.Header().Set("Content-Type", contentType)
-	t := p.templates[key]
+	if contentType == ContentTypeHTML {
+		t := p.htmlTemplates[key]
+		if t == nil {
+			return respondErr(w, http.StatusInternalServerError,
+				errors.New("no template found for "+key))
+		}
+
+		if err := t.Execute(w, values); err != nil {
+			return http.StatusInternalServerError,
+				errors.WithMessage(err, "failed to write response")
+		}
+
+		return status, nil
+	}
+
+	t := p.textTemplates[key]
```

Comment:
similar snippet is present above as well. Should we create a common function for both?

Reply:
@ayusht2810 Created a common function